File: ProjetJavax/src/entity/Stock.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity;

import java.sql.Date;
import java.util.Objects;

/**
 *
 * @author Khadija
 */
public class Stock {

    private int idS;
    private String nomPartenaireS;
    private String nomS;
    private int refS;
    private String categorieS;
    private int qteS;
    private Date dateS;
    private String qualiteS ;
    

    public Stock(String nomPartenaireS, String nomS, int refS, String categorieS, int qteS, Date dateS, String qualiteS) {
        this.nomPartenaireS = nomPartenaireS;
        this.nomS = nomS;
        this.refS = refS;
        this.categorieS = categorieS;
        this.qteS = qteS;
        this.dateS = dateS;
        this.qualiteS = qualiteS;
    }

    

    
    public Stock(int idS, String nomPartenaireS, String nomS, int refS, String categorieS, int qteS, Date dateS, String qualiteS) {
        this.idS = idS;
        this.nomPartenaireS = nomPartenaireS;
        this.nomS = nomS;
        this.refS = refS;
        this.categorieS = categorieS;
        this.qteS = qteS;
        this.dateS = dateS;
        this.qualiteS = qualiteS;
    }

  

    public Stock(String nomPartenaireS, String nomS, int refS, String categorieS, int qteS, Date dateS) {

        this.nomPartenaireS = nomPartenaireS;
        this.nomS = nomS;
        this.refS = refS;
        this.categorieS = categorieS;
        this.qteS = qteS;
        this.dateS = dateS;
    }

    

    public int getIdS() {
        return idS;
    }

    public void setIdS(int idS) {
        this.idS = idS;
    }

    public String getNomPartenaireS() {
        return nomPartenaireS;
    }

    public void setNomPartenaireS(String Partenaire) {
        this.nomPartenaireS = nomPartenaireS;
    }

    
    public String getNomS() {
        return nomS;
    }

    public void setNomS(String nomS) {
        this.nomS = nomS;
    }

    public int getRefS() {
        return refS;
    }

    public void setRefS(int refS) {
        this.refS = refS;
    }

    public String getCategorieS() {
        return categorieS;
    }

    public void setCategorieS(String categorieS) {
        this.categorieS = categorieS;
    }

    public int getQteS() {
        return qteS;
    }

    public void setQteS(int qteS) {
        this.qteS = qteS;
    }

    public Date getDateS() {
        return dateS;
    }

    public void setDateS(Date dateS) {
        this.dateS = dateS;
    }

    public String getQualiteS() {
        return qualiteS;
    }

    public void setQualiteS(String qualite) {
        this.qualiteS = qualiteS;
    }

    @Override
    public int hashCode() {
        int hash = 3;
        hash = 97 * hash + this.idS;
        hash = 97 * hash + Objects.hashCode(this.nomPartenaireS);
        hash = 97 * hash + Objects.hashCode(this.nomS);
        hash = 97 * hash + this.refS;
        hash = 97 * hash + Objects.hashCode(this.categorieS);
        hash = 97 * hash + this.qteS;
        hash = 97 * hash + Objects.hashCode(this.dateS);
        hash = 97 * hash + Objects.hashCode(this.qualiteS);
        return hash;
    }

 

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Stock other = (Stock) obj;
        if (this.idS != other.idS) {
            return false;
        }
        if (this.refS != other.refS) {
            return false;
        }
        if (this.qteS != other.qteS) {
            return false;
        }
        if (this.qualiteS != other.qualiteS) {
            return false;
        }
        if (!Objects.equals(this.nomPartenaireS, other.nomPartenaireS)) {
            return false;
        }
        if (!Objects.equals(this.nomS, other.nomS)) {
            return false;
        }
        if (!Objects.equals(this.categorieS, other.categorieS)) {
            return false;
        }
        if (!Objects.equals(this.dateS, other.dateS)) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "Stock{" + "idS=" + idS + ", nomPartenaireS=" + nomPartenaireS + ", nomS=" + nomS + ", "
                + "refS=" + refS + ", categorieS=" + categorieS + ", qteS=" + qteS + ", dateS=" + dateS + ", qualiteS=" + qualiteS + '}';
    }

    
    
   

   

  
    


}

File: DemoProject3/src/Entity/Stock.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * Stock
 *
 * @ORM\Table(name="stock", indexes={@ORM\Index(name="nomPartenaireS", columns={"nomPartenaireS"})})
 * @ORM\Entity(repositoryClass=App\Repository\StockRepository::class)
 */
class Stock
{
    /**
     * @var int
     *
     * @ORM\Column(name="idS", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $ids;

    /**
     * @var string
     * @Assert\NotBlank(message="Le NOM du Stock doit etre non vide")
     * @Assert\Type(type="alpha", message="Le nom ne doit pas contenir des chiffres .")
     * @Assert\Length(
     *      max = 8,
     *      maxMessage=" Très long !"
     *
     *     )
     * @ORM\Column(name="nomS", type="string", length=50, nullable=false)
     */
    private $noms;

    /**
     * @var string
     * @Assert\NotBlank(message="Entrez quelques choses !")
     * @Assert\Positive(message="La Matricule doit etre positif.")
     * @Assert\Range(
     *      min = 1,
     *      max = 8000,
     *      notInRangeMessage = "la Matricule doit etre valide",
     *     )
     * @ORM\Column(name="refS", type="string", length=255, nullable=false)
     */
    private $refs;

    /**
     * @var string
     *
     * @ORM\Column(name="categorieS", type="string", length=50, nullable=false)
     */
    private $categories;

    /**
     * @var int
     *  @Assert\NotBlank(message="Entrez quelques choses !")
     * @Assert\Positive(message="La Matricule doit etre positif.")
     * @Assert\Range(
     *      min = 1,
     *      max = 8000,
     *      notInRangeMessage = "la Matricule doit etre valide",
     *     )
     * @ORM\Column(name="qteS", type="integer", nullable=false)
     */
    private $qtes;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateS", type="date", nullable=false)
     */
    private $dates;

    /**
     * @var string
     *
     * @ORM\Column(name="qualiteS", type="string", length=50, nullable=false, options={"default"="1.0"})
     */
    private $qualites = '1.0';

    /**
     * @ORM\ManyToOne(targetEntity="Partenaires")
     * @ORM\JoinColumn(name="nomPartenaireS", referencedColumnName="nomMarqueP")
     
     */
    private $nompartenaires;

    /**
     * @ORM\ManyToOne(targetEntity=Partenaires::class)
     * @ORM\JoinColumn(nullable=false,referencedColumnName="idP")
     */
    private $partenaire;

    public function getIds(): ?int
    {
        return $this->ids;
    }

    public function getNoms(): ?string
    {
        return $this->noms;
    }

    public function setNoms(string $noms): self
    {
        $this->noms = $noms;

        return $this;
    }

    public function getRefs(): ?string
    {
        return $this->refs;
    }

    public function setRefs(string $refs): self
    {
        $this->refs = $refs;

        return $this;
    }

    public function getCategories(): ?string
    {
        return $this->categories;
    }

    public function setCategories(string $categories): self
    {
        $this->categories = $categories;

        return $this;
    }

    public function getQtes(): ?int
    {
        return $this->qtes;
    }

    public function setQtes(int $qtes): self
    {
        $this->qtes = $qtes;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->dates;
    }

    public function setDateS(\DateTimeInterface $dates): self
    {
        $this->dates = $dates;

        return $this;
    }

    public function getQualiteS(): ?string
    {
        return $this->qualites;
    }

    public function setQualiteS(string $qualites): self
    {
        $this->qualites = $qualites;

        return $this;
    }

    public function getNompartenaires(): ?Partenaires
    {
        return $this->nompartenaires;
    }

    public function setNompartenaires(?Partenaires $nompartenaires): self
    {
        $this->nompartenaires = $nompartenaires;

        return $this;
    }

    public function getPartenaire(): ?Partenaires
    {
        return $this->partenaire;
    }

    public function setPartenaire(?Partenaires $partenaire): self
    {
        $this->partenaire = $partenaire;

        return $this;
    }

    public function getDates(): ?\DateTimeInterface
    {
        return $this->dates;
    }


}
